test/jdk/java/net/httpclient/HttpResponseLimitingTest.java

```diff
@@ -0,0 +1,273 @@
+        HttpResponse<byte[]> response = requestBytes(version, secure, sufficientCapacity);
+        assertArrayEquals(RESPONSE_BODY, response.body());
+    }
+
+    static Arguments[] sufficientCapacities() {
+        return capacityArgs(Long.MAX_VALUE, RESPONSE_BODY.length);
+    }
+
+    @ParameterizedTest
+    @MethodSource("insufficientCapacities")
+    void testFailureOnInsufficientCapacity(HttpClient.Version version, boolean secure, long insufficientCapacity) {
+        assertThrows(
+                IOException.class,
+                () -> requestBytes(version, secure, insufficientCapacity),
+                "body exceeds capacity: " + RESPONSE_BODY.length);
+    }
+
+    static Arguments[] insufficientCapacities() {
+        return capacityArgs(0, RESPONSE_BODY.length - 1);
+    }
+
+    private static Arguments[] capacityArgs(long... capacities) {
+        return Stream
+                .of(HttpClient.Version.HTTP_1_1, HttpClient.Version.HTTP_2)
+                .flatMap(version -> Stream
+                        .of(true, false)
+                        .flatMap(secure -> Arrays
+                                .stream(capacities)
+                                .mapToObj(capacity -> Arguments.of(version, secure, capacity))))
+                .toArray(Arguments[]::new);
+    }
+
+    private static HttpResponse<byte[]> requestBytes(
+            HttpClient.Version version,
+            boolean secure,
+            long capacity)
+            throws Exception {
+
+        // Create the server and the request URI
+        SSLContext sslContext;
+        HttpServerAdapters.HttpTestServer server;
+        String handlerPath = "/";
+        URI requestUri;
+        if (secure) {
+            sslContext = new SimpleSSLContext().get();
+            server = HttpServerAdapters.HttpTestServer.create(version, sslContext);
+            requestUri = URI.create("https://" + server.serverAuthority() + handlerPath);
+        } else {
+            sslContext = null;
+            server = HttpServerAdapters.HttpTestServer.create(version);
+            requestUri = URI.create("http://" + server.serverAuthority() + handlerPath);
+        }
+
+        // Register the request handler
+        server.addHandler(
+                (exchange) -> {
+                    exchange.sendResponseHeaders(200, RESPONSE_BODY.length);
+                    try (var outputStream = exchange.getResponseBody()) {
+                        outputStream.write(RESPONSE_BODY);
+                    }
+                    exchange.close();
+                },
+                handlerPath);
+
+        // Start the server and the client
+        server.start();
+        try (var client = createClient(sslContext)) {
+
+            // Issue the request
+            var request = HttpRequest
+                    .newBuilder(requestUri)
+                    .timeout(Duration.ofSeconds(5))
```

Comment:
Given the context of this test class, I think configuring the request with a timeout shouldn't be necessary. In general, we rarely use timeouts (hardcoded ones specifically) in our tests because these tests run on a variety of hosts (some of which are slow or are running too many tests concurrently) and thus there's no right value for the timeout and can lead to intermittent failures.

Reply:
Removed the timeouts in e10771317ef57f80d25be58df77eb5c5b0083b0c.